code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -136,8 +140,43 @@ namespace <#= qualifiedNamespace #>
                     entityManager.AddComponentData(entity, updates);
                 }
 
-                // TODO: Events?
+<# if (eventDetailsList.Count > 0) { #>
+                // Events
+                var eventsObject = op.Update.SchemaData.Value.GetEvents();
+<# foreach (var eventDetails in eventDetailsList) { #>
+                {
+                    var eventCount = eventsObject.GetObjectCount(<#= eventDetails.EventIndex #>);
+                    if (eventCount > 0)
+                    {
+                        // Create component to hold received events
+                        ReceivedEvents.<#= eventDetails.PascalCaseName #> eventsReceived;
+                        List<<#= eventDetails.FullyQualifiedPayloadTypeName #>> eventList;
+                        if (!entityManager.HasComponent<ReceivedEvents.<#= eventDetails.PascalCaseName #>>(entity))
+                        {
+                            eventsReceived = new ReceivedEvents.<#= eventDetails.PascalCaseName #>() {
+                                handle = ReferenceTypeProviders.<#= eventDetails.PascalCaseName #>Provider.Allocate(World)
+                            };
+                            eventList = new List<<#= eventDetails.FullyQualifiedPayloadTypeName #>>((int) eventCount);
+                            ReferenceTypeProviders.<#= eventDetails.PascalCaseName #>Provider.Set(eventsReceived.handle, eventList);
+                            entityManager.AddComponentData(entity, eventsReceived);
+                        }
+                        else
+                        {
+                            eventsReceived = entityManager.GetComponentData<ReceivedEvents.<#= eventDetails.PascalCaseName #>>(entity);
+                            eventList = eventsReceived.Events;
+                        }
+
+                        // Deserialize events onto component
+                        for (var i = 0; i < eventCount; i++)
+                        {
+                            var e = <#= eventDetails.FullyQualifiedPayloadTypeName #>.Serialization.Deserialize(eventsObject.GetObject(<#= eventDetails.EventIndex #>));
+                            eventList.Add(e);
```

Comment:
I'd be tempted to just inline this - but it doesn't really matter since its generated code